Formula/postgresql.rb

```diff
@@ -100,18 +100,20 @@ def post_install
   end
 
   def caveats; <<-EOS.undent
-    If builds of PostgreSQL 9 are failing and you have version 8.x installed,
+    If builds of PostgreSQL 10 are failing and you have version 8.x installed,
     you may need to remove the previous version first. See:
       https://github.com/Homebrew/legacy-homebrew/issues/2510
 
     To migrate existing data from a previous major version (pre-9.0) of PostgreSQL, see:
-      https://www.postgresql.org/docs/9.6/static/upgrading.html
+      https://www.postgresql.org/docs/10/static/upgrading.html
 
-    To migrate existing data from a previous minor version (9.0-9.5) of PostgreSQL, see:
-      https://www.postgresql.org/docs/9.6/static/pgupgrade.html
+    To migrate existing data from a previous minor version (9.0-9.6) of PostgreSQL, see:
```

Comment:
As there weren't any minor version of `10.0`, should it be substituted for `10.0 Beta1-4`?

Reply:
Actually the comment is wrong, 9.0, 9.1, 9.2 etc. are all *major* versions, while 9.6.4 to 9.6.5 would be a minor version update, which doesn't require any changes to the database.

In Postgres 10 the versioning schema has changed, and now 10 and 11 would be major versions, but 10.0 and 10.1 would be minor versions.

Quoting https://www.postgresql.org/docs/10/static/upgrading.html:

> Current PostgreSQL version numbers consist of a major and a minor version number. For example, in the version number 10.1, the 10 is the major version number and the 1 is the minor version number, meaning this would be the first minor release of the major release 10. For releases before PostgreSQL version 10.0, version numbers consist of three numbers, for example, 9.5.3. In those cases, the major version consists of the first two digit groups of the version number, e.g., 9.5, and the minor version is the third number, e.g., 3, meaning this would be the third minor release of the major release 9.5.
> 
> Minor releases never change the internal storage format and are always compatible with earlier and later minor releases of the same major version number. For example, version 10.1 is compatible with version 10.0 and version 10.6. Similarly, for example, 9.5.3 is compatible with 9.5.0, 9.5.1, and 9.5.6. To update between compatible versions, you simply replace the executables while the server is down and restart the server. The data directory remains unchanged — minor upgrades are that simple.

So really the comment should have said "to migrate data from a previous major version".

Beta versions are generally excluded from normal update rules, because they might contain braking changes in the data format that are later reverted, so to upgrade from a beta to a release version often requires an upgrade using full dump and restore.

I don't think we need to mention beta versions here at all, since homebrew didn't release beta versions of postgres as far as I know.

Reply:
:+1: